src/main/sensors/gyro_init.c

```diff
@@ -577,23 +570,19 @@ bool gyroInit(void)
         break;
     }
 
-    gyroDetectionFlags = GYRO_NONE_MASK;
+    gyroDetectionFlags = 0;
     uint8_t gyrosToScan = gyroConfig()->gyrosDetected;
 
-    gyro.gyroToUse = gyroConfig()->gyro_to_use;
+    gyro.gyroEnabledBitmask = gyroConfig()->gyro_enabled_bitmask;
     gyro.gyroDebugAxis = gyroConfig()->gyro_filter_debug_axis;
 
-    if ((!gyrosToScan || (gyrosToScan & GYRO_1_MASK)) && gyroDetectSensor(&gyro.gyroSensor1, gyroDeviceConfig(0))) {
-        gyroDetectionFlags |= GYRO_1_MASK;
-    }
-
-#if defined(USE_MULTI_GYRO)
-    if ((!gyrosToScan || (gyrosToScan & GYRO_2_MASK)) && gyroDetectSensor(&gyro.gyroSensor2, gyroDeviceConfig(1))) {
-        gyroDetectionFlags |= GYRO_2_MASK;
+    for (int i = 0; i < GYRO_COUNT; i++) {
+        if ((!gyrosToScan || (gyrosToScan & GYRO_MASK(i))) && gyroDetectSensor(&gyro.gyroSensor[i], gyroDeviceConfig(i))) {
+            gyroDetectionFlags |= GYRO_MASK(i);
```

Comment:
(this will mark all gyros as detected when gyrosToScan ==0 ? )

Reply:
Do we even need gyrosToScan? It was part of the old code, but couldn't I just scan each gyro? Any reason why we have gyroConfig()->gyrosDetected? That is currently only used to check which gyros to scan, and if its set to 0, it should just scan all gyros. So I'm thinking of just removing this logic and scanning all IMU and just using those that are detected, what do you think?

Reply:
Sorry, I did not check related code.
https://github.com/betaflight/betaflight/pull/9675

IMO main problem is that gyro detection is taking so long. But boot time difference is huge, so some mechanism should be implemented, at least until detection time is improved.

Maybe the logic can be:
- reset value 0xff
- if mask == 0xff -> update with detection result
- if mask == 0 -> detect all gyros, but do not modify mask

Reply:
Just looking at this again, the old logic actually seems correct. If gyrosToScan ==0 then it will just check all gyros and try to find which gyros are detected. It does not automatically mark them as detected, but it will check if they are detected.

Reply:
@ledvinap is this resolved, or how should this logic of detecting gyros be done?